src/test/java/net/minestom/server/collision/EntityBlockTouchTickIntegrationTest.java

```diff
@@ -38,37 +39,106 @@ public void onTouch(@NotNull Touch touch) {
+        entity.setInstance(instance, spawnPoint).join();
+
+        Arrays.stream(Direction.values())
+                .filter(direction -> direction != Direction.DOWN)
+                .map(Direction::vec)
+                .forEachOrdered(vec -> {
+            entity.setVelocity(vec.mul(10));
+            entity.tick(0);
+            entity.teleport(spawnPoint);
+        });
+
+        assertEquals(blockPos, positions);
+
+        positions.clear(); // Final -Y check
+        instance.setBlock(0, 42, 0, customBlock);
         entity.tick(0);
+        assertEquals(Set.of(new Vec(0, 42, 0)), positions);
 
-        assertEquals(Set.of(new Vec(0, 42, 0),
-                new Vec(0, 42, 1),
-                new Vec(0, 43, 1)),
-                positions);
+        assertEquals(instance, entity.getInstance());
+    }
+    @Test
+    public void entityPhysicsCheckTouchTickFarPositiveXNegativeZ(Env env) {
+        var instance = env.createFlatInstance();
+        instance.loadChunk(new Pos(1000, 1000, -1000));
+
+        Set<Point> positions = new HashSet<>();
+        var handler = new BlockHandler() {
+            @Override
+            public void onTouch(@NotNull Touch touch) {
+                assertTrue(positions.add(touch.blockPosition()));
+            }
+
+            @Override
+            public @NotNull NamespaceID getNamespaceId() {
+                return NamespaceID.from("minestom:test");
+            }
+        };
+
+        var customBlock = Block.STONE.withHandler(handler);
+        var offset = new Vec(1000, 0, -1000);
+        var blockPos = Set.of(offset.add(-1, 43, 0),
+                offset.add(1, 43, 0),
+                offset.add(0, 43, -1),
+                offset.add(0, 43, 1),
+                offset.add(0, 45, 0) // Y+
+        );
+
+        for (var pos : blockPos) {
+            instance.setBlock(pos, customBlock);
+        }
+        instance.setBlock(1000, 42, -1000, Block.STONE); // Regular floor as we are going to be sliding into the blocks.
+
+        var entity = new Entity(EntityType.ZOMBIE);
+        var spawnPoint = new Pos(1000.5, 43, -999.5);
+        entity.setInstance(instance, spawnPoint).join();
+
+        Arrays.stream(Direction.values())
+                .filter(direction -> direction != Direction.DOWN)
+                .map(Direction::vec)
+                .forEachOrdered(vec -> {
+                    entity.setVelocity(vec.mul(10));
+                    entity.tick(0);
+                    entity.teleport(spawnPoint);
+                });
+
+        assertEquals(blockPos, positions);
+
+        positions.clear(); // Final -Y check
+        instance.setBlock(1000, 42, -1000, customBlock);
+        entity.tick(0);
+        assertEquals(Set.of(new Vec(1000, 42, -1000)), positions);
 
         assertEquals(instance, entity.getInstance());
     }
 
     @Test
-    public void entityPhysicsCheckTouchTickFarZ(Env env) {
+    public void entityPhysicsCheckTouchTickFarNegativeXPositiveZ(Env env) {
```

Comment:
Nitpick: Remove public for junit5.
Ref: [junit.org/junit5/docs/snapshot/user-guide#writing-tests-classes-and-methods](https://junit.org/junit5/docs/snapshot/user-guide/#writing-tests-classes-and-methods)